modo/api.py

```diff
@@ -57,7 +59,15 @@ def __init__(
         has_assay: List = [],
         source_uri: Optional[str] = None,
     ):
-        self.path: Path = Path(path)
+        if is_uri(path):
+            s3_endpoint, modo_path = split_s3_url(path)
+            archive = zarr.open_group(
+                f"s3://{modo_path}/data.zarr",
+                storage_options={"anon": True, "endpoint_url": s3_endpoint},
+            )
+        else:
+            self.path = Path(path)
```

Comment:
**praise:** neat! I didn't realize you could do it so easily with zarr :)
**question:** this is simpler than what we use in modo_server (below). Would it make sense to  use the same pattern in there? Could we even remove the `archive` of `MODO` altogether?

https://github.com/sdsc-ordes/modo-api/blob/46ce93862bd6caab73d1ec65c89fab8735c1b7f6/deploy/modo-server/server.py#L42-L46